functions/jsonwebtoken.js

```diff
@@ -0,0 +1,33 @@
+require('dotenv').config();
+const jwt = require('jsonwebtoken');
+const getTokenFrom = require('./get-token-from');
+
+module.exports = {
+  generateAccesToken: (user) => {
+    if (!user.role || !user.id || !user.username) return null; // access denied, it returns null
+
+    const userForToken = {
+      username: user.username,
+      role: user.role,
```

Comment:
We have a `Role` but it's the full association which have every property of Role's model. Wouldn't it be better if we store only the `roleId`?

Reply:
I think so, with the id Role we can verify what kind of user make the request, we don't need the role name in this case.

Reply:
Gonza, as we have talked about, I thought that the function generateAccessToken would receive an object with those keys and not the complete user data. However, no problem! I'll modify it in a minute.